File: AtlasWeb/Core/Modules/Tibbi_Surec/Tibbi_Surec_Evrensel_Modulu/AppointmentFormBaseViewModel.ts
//$D0EF672B
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { BaseAction } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class AppointmentFormBaseViewModel extends BaseViewModel {
    @Type(() => BaseAction)
    public _BaseAction: BaseAction = new BaseAction();

    public AppointmentDescription: string;
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Tibbi_Surec_Evrensel_Modulu/AppointmentFormBaseViewModel.cs
//$E2651808
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class BaseActionServiceController
    {
        partial void PreScript_AppointmentFormBase(AppointmentFormBaseViewModel viewModel, BaseAction baseAction, TTObjectContext objectContext)
        {
            viewModel.AppointmentDescription = BaseAction.GetFullAppointmentDescription(baseAction);
        }

        partial void PostScript_AppointmentFormBase(AppointmentFormBaseViewModel viewModel, BaseAction baseAction, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            baseAction.CompleteMyNewAppoinments();
        }
    }
}

namespace Core.Models
{
    public partial class AppointmentFormBaseViewModel
    {
        public string AppointmentDescription;
    }
}